pkg/controller/bundle/bundle_unpacker.go

```diff
@@ -342,7 +356,22 @@ func (c *ConfigMapUnpacker) UnpackBundle(lookup *operatorsv1alpha1.BundleLookup)
 		return
 	}
 
-	if !jobConditionTrue(job, batchv1.JobComplete) && (cond.Status != corev1.ConditionTrue || cond.Reason != JobIncompleteReason) {
+	// Check if bundle unpack job has failed due a timeout
+	// Return a BundleJobError so we can mark the InstallPlan as Failed
+	isFailed, jobCond := jobConditionTrue(job, batchv1.JobFailed)
+	if isFailed {
+		cond.Status = corev1.ConditionTrue
+		cond.Reason = jobCond.Reason
+		cond.Message = jobCond.Message
+		cond.LastTransitionTime = &now
+		result.SetCondition(cond)
+
+		err = NewBundleJobError(fmt.Sprintf("Bundle extract Job failed with Reason: %v, and Message: %v", jobCond.Reason, jobCond.Message))
```

Comment:
Yeah that should be possible. Although I probably need to introduce a new bundleLookup.status condition type e.g `BundleLookupFailed`. So I can set that when `UnpackBundle()` sees that the Job has failed. And then in the main `syncInstallPlans()` loop I can check for the failed bundleLookup status condition to set the InstallPlan to failed as well.

Reply:
Or actually right now we have `BundleLookupPending.status=true` indicating that we're still waiting for bundle unpacking.
The absence of the `BundleLookupPending` condition indicates a successful unpack of the bundle.
`BundleLookupPending.status=false` isn't being used to signify anything so I could use that to indicate a bundle unpack failure.

So I probably don't need a new condition type for failure.

Reply:
I don't think that Pending being absent or False indicates anything about the failure state -- that should be absent or False for successfully completed lookups too.

Reply:
Yeah in this current update I ended up recycling `BundleLookupPending.status=False` since it was unused. But I agree that's probably not as clear. I'm going to update with a dedicated failure condition e.g `BundleLookupFailed`.